oxenss/storage/database.cpp

```diff
@@ -798,6 +822,24 @@ std::vector<std::string> Database::delete_by_timestamp(
     return get_all<std::string>(st, pubkey, to_epoch_ms(timestamp), ns);
 }
 
+void Database::revoke_subkey(
+        const user_pubkey_t& pubkey, std::array<unsigned char, 32> revoke_subkey) {
+    auto get_owner = impl->prepared_st("SELECT id FROM owners WHERE pubkey = ? AND type = ?");
+    auto insert_subkey = impl->prepared_st(
+            "INSERT INTO revoked_subkeys (owner, subkey) VALUES (?, ?) ON CONFLICT DO NOTHING "
+            "RETURNING id");
+    auto ownerid = exec_and_get<int64_t>(get_owner, pubkey);
+    exec_and_get<int64_t>(insert_subkey, ownerid, (std::string)util::view_guts(revoke_subkey));
+}
+
+bool Database::subkey_revoked(std::array<unsigned char, 32> revoke_subkey) {
+    auto get_subkey_count =
+            impl->prepared_st("SELECT count(*) FROM revoked_subkeys WHERE subkey = ?");
+    auto subkey_count = exec_and_maybe_get<int64_t>(
```

Comment:
```suggestion
    auto subkey_count = exec_and_get<int64_t>(
```
`exec_and_maybe_get` is for when it might return 0 or 1 rows and wraps it in an `std::optional`; this query *always* returns exactly 1 result, so no need to go through the std::optional.